File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0006-join/type/TDeptTable.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tDeptTable"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public interface TDeptTable extends com.gs.dmn.runtime.DMNType {
    static TDeptTable toTDeptTable(Object other) {
        if (other == null) {
            return null;
        } else if (TDeptTable.class.isAssignableFrom(other.getClass())) {
            return (TDeptTable)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TDeptTableImpl result_ = new TDeptTableImpl();
            result_.setNumber((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("number"));
            result_.setName((String)((com.gs.dmn.runtime.Context)other).get("name"));
            result_.setManager((String)((com.gs.dmn.runtime.Context)other).get("manager"));
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTDeptTable(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TDeptTable.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("number")
    java.math.BigDecimal getNumber();

    @com.fasterxml.jackson.annotation.JsonGetter("name")
    String getName();

    @com.fasterxml.jackson.annotation.JsonGetter("manager")
    String getManager();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("number", getNumber());
        context.put("name", getName());
        context.put("manager", getManager());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TDeptTable other = (TDeptTable) o;
        if (this.getManager() != null ? !this.getManager().equals(other.getManager()) : other.getManager() != null) return false;
        if (this.getName() != null ? !this.getName().equals(other.getName()) : other.getName() != null) return false;
        if (this.getNumber() != null ? !this.getNumber().equals(other.getNumber()) : other.getNumber() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getManager() != null ? this.getManager().hashCode() : 0);
        result = 31 * result + (this.getName() != null ? this.getName().hashCode() : 0);
        result = 31 * result + (this.getNumber() != null ? this.getNumber().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("manager=" + getManager());
        result_.append(", name=" + getName());
        result_.append(", number=" + getNumber());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0006-join/type/TDeptTable.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tDeptTable"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TDeptTableImpl::class)
interface TDeptTable : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("number")
    val number: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("name")
    val name: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("manager")
    val manager: String?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("number", this.number)
        context.put("name", this.name)
        context.put("manager", this.manager)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TDeptTable
        if (if (this.manager != null) this.manager != other.manager else other.manager != null) return false
        if (if (this.name != null) this.name != other.name else other.name != null) return false
        if (if (this.number != null) this.number != other.number else other.number != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.manager != null) this.manager.hashCode() else 0)
        result = 31 * result + (if (this.name != null) this.name.hashCode() else 0)
        result = 31 * result + (if (this.number != null) this.number.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("manager=" + manager)
        result_.append(", name=" + name)
        result_.append(", number=" + number)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTDeptTable(other: Any?): TDeptTable? {
            if (other == null) {
                return null
            } else if (other is TDeptTable?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TDeptTableImpl()
                result_.number = other.get("number") as java.math.BigDecimal?
                result_.name = other.get("name") as String?
                result_.manager = other.get("manager") as String?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTDeptTable(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TDeptTable::class.java.getSimpleName()))
            }
        }
    }
}
